addons/mail/static/src/new/thread/message.js

```diff
@@ -159,6 +159,26 @@ export class Message extends Component {
         return this.messaging.state.threads[this.props.threadLocalId].id === "inbox";
     }
 
+    /**
+     * @returns {boolean}
+     */
+    get shouldDisplayAuthorName() {
+        if (!this.env.inChatWindow) {
+            return true;
+        }
+        if (this.message.isAuthoredByCurrentUser) {
+            return false;
+        }
+        if (this.thread.type === "chat") {
+            return false;
+        }
+        return true;
+    }
+
+    get thread() {
+        return this.messaging.state.threads[this.props.threadLocalId];
+    }
```

Comment:
`props.threadLocalId` is an optional props, you are missing a guard here.

Reply:
In fact, this could be in the setup and use the guard on this.thread instead of the props. Would be cleaner